Add prediction entropy for membership inference attack

lwsong opened the pull request:
Our recent paper https://arxiv.org/pdf/2003.10595.pdf shows that conventional or modified prediction entropy can also be used to perform successful thresholding-based membership inference attacks. Here, I implement the code to compute entropy value in "AttackInputData" Class in "tensorflow_privacy/privacy/membership_inference_attack/data_structures.py"

lwsong commented:
@googlebot I signed it!

CdavM commented:
Hi @lwsong ! Thanks for the PR. 

Can you clarify what the changes to the codelab are? I think github messed up the file diff and I'm having a hard time parsing it.

Also, since you're the only one familiar with the definition of entropy, can you add a few tests for these functions (to data_structures_test.py)? This will help us make sure that they keep working as intended.

Thanks!
David

lwsong commented:
Hi @CdavM , for the codelab, I just changed two lines to calculate the probabilities by specifying the axis: 
prob_train = special.softmax(logits_train, axis=1)
prob_test = special.softmax(logits_test, axis=1)
I also added a few entropy tests to data_structures_test.py.

Regards,
Liwei
